sql_grader/mixins/grading.py

```diff
@@ -82,19 +133,104 @@ def _calculate_score(self):
         """
         raw_possible = 1.0
         raw_earned = 0.0
-        database = DATABASES[self.dataset]
-        problem = SqlProblem(
-            database,
+        actual, expected, error, comparison = attempt_safe(
+            self.dataset,
             self.answer_query,
             self.verify_query,
             self.is_ordered,
+            self.raw_response
         )
-        response = self.raw_response
-        actual, expected, error, comparison = problem.attempt(response)
         if comparison:
             raw_earned = 1.0
         score = Score(
             raw_earned=raw_earned,
             raw_possible=raw_possible,
         )
         return (score, actual, expected, error, comparison)
+
+
+class XBlockDataMixin:
```

Comment:
FWIW, I'm not really clear on why this class was moved, but I have no real opinion about it one way or the other.

Reply:
I moved it because it depends on xblock imports. Since codejail is strict about what modules are available, I refactored the `models.py` to only import stdlib stuff, rather than having to copy in all of the xblock libraries and etc.

Reply:
Makes sense why we need to rearrange.

Given that, can we rename `models.py` to `problem.py` or some such?

All Stanford-derived XBlocks mimic the djangoapp convention that "`models.py` stores datastore-backed stuff".

It'd be nice to maintain that expectation. Historically, the file was originally called `problem.py` (when it was separated from the datastore stuff), prior to pre-merge refactor.